pkgs/by-name/ba/batmon/package.nix

```
@@ -0,0 +1,32 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "batmon";
+  version = "0.0.1";
+
+  src = fetchFromGitHub {
+    owner = "6543";
+    repo = "batmon";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-+kjDNQKlaoI5fQ5FqYF6IPCKeE92WKxIhVCKafqfE0o=";
+  };
+
+  cargoSha256 = "sha256-DJpWBset6SW7Ahg60+Tu1VpH34LcVOyrEs9suKyTE9g=";
+
+  meta = with lib; {
+    description = "Interactive batteries viewer";
+    longDescription = ''
+      An interactive viewer, similar to top, htop and other *top utilities,
+      but about the batteries installed in your notebook.
+    '';
+    homepage = "https://github.com/6543/batmon/";
+    changelog = "https://github.com/6543/batmon/releases/tag/v${version}";
+    license = licenses.asl20;
+    mainProgram = "batmon";
+    platforms = platforms.linux;
```

If it should be usable on darwin, you should probably modify this to include it.

---

I dont have a way to test it, feel free to push to it @h7x4

---

I am not able to push to it, as the `Allow edits from maintainers` box seems to be unchecked. But I had a look upstream, and found out that `starship-battery` supports a whole range of systems.

```suggestion
    platforms = with platforms; unix ++ windows;
```

---

Yes but the darwin builder (x86) failed and some told me it does not work on the darwin-arm arch ...

---

I can apply but we will see the builder fail again ... :/

---

If `aarch64-darwin` is broken, you should set `meta.broken = stdenv.isDarwin && stdenv.isAarch64;`. `meta.platforms` is usually meant to tell you what platforms the program is meant for and supports upstream, not what we are able to build.